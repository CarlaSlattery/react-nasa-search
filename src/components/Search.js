import React, { useState } from "react";
import PropTypes from "prop-types";
import "../styles/search.css";
import getImages from "../requests/getImages.js";

const Search = ({ setSearchResults, setLoading }) => {
  const [value, setValue] = useState();

  const handleSubmit = async (event) => {
    event.preventDefault();
    setSearchResults(await getImages(value));
    setLoading(false);
  };
  return (
    <>
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="search-input"
          type="text"
          onChange={(e) => setValue(e.target.value)}
        />
        <button className="search-btn" type="submit">
          Go!
        </button>
      </form>
    </>
  );
};

export default Search;

Search.propTypes = {
  setSearchResults: PropTypes.func.isRequired,
};
