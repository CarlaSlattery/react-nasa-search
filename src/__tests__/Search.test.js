import { fireEvent, render, screen } from "@testing-library/react";
import Search from "../components/Search";

const Button = ({ onSubmit }) => <button onSubmit={onSubmit}></button>;

describe("Search", () => {
  const setSearchResults = jest.fn();

  it("renders correctly", () => {
    const { asFragment } = render(
      <Search setSearchResults={setSearchResults} />
    );
    expect(asFragment()).toMatchSnapshot();

    const searchElement = screen.getByRole("textbox");
    expect(searchElement).toBeInTheDocument();

    const searchButton = screen.getByRole("button");
    expect(searchButton).toBeInTheDocument();
  });
});
