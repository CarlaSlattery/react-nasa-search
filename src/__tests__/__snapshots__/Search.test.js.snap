// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search-form"
  >
    <input
      class="search-input"
      placeholder="Enter your search"
      type="text"
    />
    <button
      class="search-btn"
      type="submit"
    >
      Go!
    </button>
  </form>
</DocumentFragment>
`;
